File: frontend/src/types/highlight.ts
export interface Highlight {
  id: string
  book_id?: string
  book_title?: string
  book_author?: string
  asin?: string
  content: string
  explanation?: string
  location?: string
  highlighted_at?: string
  source: 'kindle'
  created_at: string
}

export interface ListBookHighlightsResponse {
  highlights: Highlight[]
}

File: backend/internal/domain/highlight.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

const (
	HighlightSourceKindle = "kindle"
)

type Highlight struct {
	ID            uuid.UUID
	UserID        uuid.UUID
	BookID        *uuid.UUID
	BookTitle     *string
	BookAuthor    *string
	ASIN          *string
	Content       string
	Explanation   *string
	ContentHash   *string
	Location      *string
	HighlightedAt *time.Time
	Source        string
	CreatedAt     time.Time
	UpdatedAt     time.Time
}
